### PR TITLE
Created Schema from Yaml to JSON 

### LINKED ISSUE #6363: Publish configuration file schema to http://schemastore.org
This will enable auto-completion & validation for the Read the Docs configuration file in various editors such as VS Code.

See http://schemastore.org

### PR DESCRIPTION
**Pull request by Blackcipher101**
fixes #6363 
Created Schema on JSON by finding matching properties in yml file https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/rtd_tests/fixtures/spec/v2/schema.yml

### CODE REVIEW

File: readthedocs/rtd_tests/fixtures/spec/v2/schema.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  "title": "Schema for Read the Docs configuration file",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "basics": {
+      "type": "object",
+      "additionalProperties": true,
+      "properties": {
+	      "version": "enum('2')",
+	      "formats": "any(list(enum('htmlzip', 'pdf', 'epub')), enum('all'), required=False)",
```
Comment (stsewd) [inline readthedocs/rtd_tests/fixtures/spec/v2/schema.json]: Believe this isn't the way of doing a list of option is json schema, and all others too (boolean values, enums, etc)

Comment (Blackcipher101) [reply readthedocs/rtd_tests/fixtures/spec/v2/schema.json]: ok I will look into it

Comment (Blackcipher101) [reply readthedocs/rtd_tests/fixtures/spec/v2/schema.json]: @stsewd Should the it go like`"properties": {
              "version": "enum['2']",
              "formats": "any(array[enum['htmlzip', 'pdf', 'epub']], enum['all'], required=False)",
              "conda": "include('conda', required=False)",
              "build": "include('build', required=False)",
`

Comment (stsewd) [reply readthedocs/rtd_tests/fixtures/spec/v2/schema.json]: No, please refer to https://json-schema.org/learn/getting-started-step-by-step.html if you are not familiar with json schema

Comment (Blackcipher101) [reply readthedocs/rtd_tests/fixtures/spec/v2/schema.json]: @stsewd I looked into it 
does the code go like this
  "properties": {
              "version":{
                      "description":"The version of the spec to be use",
                      "type":"string",
                      "enum":["2"]
              },

**Comment by dave-shawley** [comment]
FWIW, I needed to do this today and came up with https://gist.github.com/dave-shawley/b9f91a7c9ce7d2fbcaef33375d5463d0 which **I believe** is a complete JSON schema based on the YAML schema.  I'm also pretty sure that "3.7" should be added [here](https://gist.github.com/dave-shawley/b9f91a7c9ce7d2fbcaef33375d5463d0#file-schema-json-L61)

**Comment by stsewd** [comment]
Sorry, this isn't a priority for core team right now to review, hope I get some time soon

**Comment by SethFalco** [comment]
It'd be nice to see it added to the catalog of [JSON Schema Store](https://www.schemastore.org/json/).
That way, editors like Visual Studio Code or IntelliJ can automatically offer code-completion for it.

Once this is merged, I'll gladly make the PR there to include it in the list.

**Pull request closed without merging**